rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1172,21 +1178,30 @@ def _combine_sparse_dense_features(
         mask: tf.Tensor,
         name: Text,
         sparse_dropout: bool = False,
+        dense_dropout: bool = False,
     ) -> tf.Tensor:
 
         dense_features = []
 
         for f in features:
             if isinstance(f, tf.SparseTensor):
                 if sparse_dropout:
-                    _f = self._tf_layers[f"sparse_dropout.{name}"](f, self._training)
+                    _f = self._tf_layers[f"sparse_input_dropout.{name}"](
+                        f, self._training
+                    )
                 else:
                     _f = f
                 dense_features.append(self._tf_layers[f"sparse_to_dense.{name}"](_f))
             else:
                 dense_features.append(f)
 
-        return tf.concat(dense_features, axis=-1) * mask
+        outputs = tf.concat(dense_features, axis=-1) * mask
+        if dense_dropout:
+            outputs = self._tf_layers[f"dense_input_dropout.{name}"](
+                outputs, self._training
+            )
```

Comment:
This would result in almost double dropout. Sparse features would be dropped out once, then transformed, concatenated with dense and then again dropped out. Is that intended?
Any empirical benefit of doing it eitherwise?

Reply:
but that's usual, there is a layer between them

Reply:
that's intended

Reply:
I guess what I am asking is that do we need a dropout on the dense version of the sparse features. Wouldn't hurt I guess.

Reply:
> Wouldn't hurt I guess.

that is my thinking 😅